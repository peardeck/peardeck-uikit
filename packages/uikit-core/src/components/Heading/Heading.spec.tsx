import React from "react";
import { create } from "react-test-renderer";
import {
  Heading,
  Heading1,
  Heading2,
  Heading3,
  Heading4,
  Heading5,
  Heading6,
} from "./Heading";

describe("Heading", () => {
  it("matches the snapshot", () => {
    const json = create(<Heading>Default Heading</Heading>).toJSON();
    expect(json).toMatchSnapshot();
    expect(json && json.type).toEqual("h1");
  });

  describe("Heading1", () => {
    it("matches the snapshot", () => {
      const json = create(<Heading1>Heading 1</Heading1>).toJSON();
      expect(json).toMatchSnapshot();
      expect(json && json.type).toEqual("h1");
    });
  });

  describe("Heading2", () => {
    it("matches the snapshot", () => {
      const json = create(<Heading2>Heading 2</Heading2>).toJSON();
      expect(json).toMatchSnapshot();
      expect(json && json.type).toEqual("h2");
    });
  });

  describe("Heading3", () => {
    it("matches the snapshot", () => {
      const json = create(<Heading3>Heading 3</Heading3>).toJSON();
      expect(json).toMatchSnapshot();
      expect(json && json.type).toEqual("h3");
    });
  });

  describe("Heading4", () => {
    it("matches the snapshot", () => {
      const json = create(<Heading4>Heading 4</Heading4>).toJSON();
      expect(json).toMatchSnapshot();
      expect(json && json.type).toEqual("h4");
    });
  });

  describe("Heading5", () => {
    it("matches the snapshot", () => {
      const json = create(<Heading5>Heading 5</Heading5>).toJSON();
      expect(json).toMatchSnapshot();
      expect(json && json.type).toEqual("h5");
    });
  });

  describe("Heading6", () => {
    it("matches the snapshot", () => {
      const json = create(<Heading6>Heading 6</Heading6>).toJSON();
      expect(json).toMatchSnapshot();
      expect(json && json.type).toEqual("h6");
    });
  });
});
