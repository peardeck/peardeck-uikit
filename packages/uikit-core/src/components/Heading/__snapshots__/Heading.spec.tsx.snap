// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Heading1 matches the snapshot 1`] = `
.c0 {
  margin: 0;
}

<h1
  className="c0"
>
  Heading 1
</h1>
`;

exports[`Heading Heading2 matches the snapshot 1`] = `
.c0 {
  margin: 0;
}

<h2
  className="c0"
>
  Heading 2
</h2>
`;

exports[`Heading Heading3 matches the snapshot 1`] = `
.c0 {
  margin: 0;
}

<h3
  className="c0"
>
  Heading 3
</h3>
`;

exports[`Heading Heading4 matches the snapshot 1`] = `
.c0 {
  margin: 0;
}

<h4
  className="c0"
>
  Heading 4
</h4>
`;

exports[`Heading Heading5 matches the snapshot 1`] = `
.c0 {
  margin: 0;
}

<h5
  className="c0"
>
  Heading 5
</h5>
`;

exports[`Heading Heading6 matches the snapshot 1`] = `
.c0 {
  margin: 0;
}

<h6
  className="c0"
>
  Heading 6
</h6>
`;

exports[`Heading matches the snapshot 1`] = `
.c0 {
  margin: 0;
}

<h1
  className="c0"
>
  Default Heading
</h1>
`;
